src/main/webapp/intrigue-api/query-templates/query-templates.js

```diff
@@ -0,0 +1,90 @@
+const ROOT = '/search/catalog/internal'
+
+import { getIn } from 'immutable'
+
+const inverseKeys = object => {
+  return Object.keys(object).reduce((ret, key) => {
+    ret[object[key]] = key
+    return ret
+  }, {})
+}
+
+const transformAttrs = {
+  accessAdministrators: 'security_access_administrators',
+  accessGroups: 'security_access_groups',
+  accessGroupsRead: 'security_access_groups_read',
+  accessIndividuals: 'security_access_individuals',
+  accessIndividualsRead: 'security_access_individuals_read',
+  created: 'created',
+  filterTemplate: 'filterTree',
+  modified: 'modified',
+  owner: 'metacard_owner',
+  querySettings: 'query_settings',
+  id: 'id',
+  title: 'title',
+}
+
+const untransformAttrs = inverseKeys(transformAttrs)
+
+const renameKeys = (f, map) => {
+  return Object.keys(map).reduce((attrs, attr) => {
+    const name = f(attr)
+    attrs[name] = map[attr]
+    return attrs
+  }, {})
+}
+
+export const getQueryTemplates = async (parent, args, { fetch }) => {
+  const res = await fetch(`${ROOT}/forms/query`)
+  const json = await res.json()
+  const attributes = json
+    .map(attrs => renameKeys(k => transformAttrs[k], attrs))
+    .map(({ modified, created, ...rest }) => {
+      return {
+        ...rest,
+        created: new Date(created).toISOString(),
+        modified: new Date(modified).toISOString(),
+      }
+    })
+  return { attributes }
+}
+
+export const saveQueryTemplate = async (parent, args, { fetch }) => {
+  const { attrs, id } = args
+  if (getIn(attrs, ['filterTree', 'filters', 'length'], 0) === 1) {
+    attrs.filterTree = attrs.filterTree.filters[0]
```

Comment:
I don't think we should update args in place.

Reply:
since we destructure attrs from args, I don't think I am modifying the actual args object right?

Reply:
The destructure is giving you a name into an existing object, it's not copying the object for you.

Reply:
I think it's okay the way it is now, but could use a sanity check?

Reply:
I think you might be able to use [setIn](https://immutable-js.github.io/immutable-js/docs/#/setIn) instead.